### PR TITLE
Make StsWebhook and PodWebhook work

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

Close https://github.com/pingcap/tidb-operator/issues/2095

Currently sts webhook and pod webhook couldn't work at the same time because when pod webhook enabled, the sts partition would direct be set as 0.

This pr fix this and make e2e test to ensure it.

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Make StatefulSet and Pod validation webhook could work at the sametime.
```

### CODE REVIEW

File: tests/e2e/tidbcluster/serial.go [hunk 1]
```diff
@@ -137,19 +138,58 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
 			klog.Info("start to upgrade tidbcluster with pod admission webhook")
 			// deploy new cluster and test upgrade and scale-in/out with pod admission webhook
-			cluster := newTidbClusterConfig(e2econfig.TestConfig, ns, "admission", "", utilimage.TiDBV3Version)
-			cluster.Resources["pd.replicas"] = "3"
-			cluster.Resources["tikv.replicas"] = "3"
-			cluster.Resources["tidb.replicas"] = "2"
+			tc := fixture.GetTidbCluster(ns, "admission", utilimage.TiDBV3Version)
+			tc.Spec.PD.Replicas = 3
+			tc.Spec.TiKV.Replicas = 3
+			tc.Spec.TiDB.Replicas = 2
```
Comment (Yisaer) [inline tests/e2e/tidbcluster/serial.go:144]: use crd instead

File: tests/e2e/tidbcluster/serial.go [hunk 2]
```diff
@@ -912,3 +952,15 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+	// add annotation to pause statefulset upgrade process
+	cmd := fmt.Sprintf("kubectl annotate tc %s -n %s tidb.pingcap.com/%s-partition=%d --overwrite",
+		tcName, namespace, component, ordinal)
+	klog.Infof("%s", cmd)
+	output, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("fail to set annotation for [%s/%s], component: %s, partition: %d, err: %v, output: %s", namespace, tcName, component, ordinal, err, string(output))
+	}
+	return nil
+}
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go:966]: use `framework.RunKubectl` or [related functions](https://github.com/kubernetes/kubernetes/blob/68ccd2e7129b9e7064e353d3dd05fcff83d5afc3/test/e2e/framework/util.go#L1557-L1591) to run `kubectl` command, e.g. 

https://github.com/pingcap/tidb-operator/blob/46ad1990137adbd7ea679aeca37de93879a55afd/tests/e2e/tidbcluster/tidbcluster.go#L1223-L1232

Comment (Yisaer) [reply tests/e2e/tidbcluster/serial.go:962]: updated.

**Pull request merged**